From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Tue, 1 Sep 2026 15:52:44 +0200
Subject: fix(busybox): do not install mke2fs applet

e2fsprogs provides `mke2fs`. `mkfs.ext2`, `mkfs.ext3`, and `mkfs.ext4`
are symlinks to it. Busybox does not provide a `mkfs.ext4` applet and
`mkfs.ext*` would point to busybox then. This can be seen when including
the busybox module:

```
$ test/test.sh ubuntu:devel 21
[...]
CLIENT TEST START: encrypted overlay (new device, random password)
[...]
[    7.398868] dracut-pre-pivot[289]: Creating ext4 filesystem on /dev/mapper/overlay-crypt
[    7.400071] dracut-pre-pivot[373]: mkfs.ext4: applet not found
```

So do not install the `mke2fs` busybox applet.

Origin: upstream, https://github.com/dracut-ng/dracut/commit/38247fbc11040521aef6f1e1b2fe494538967d50
---
 modules.d/10busybox/module-setup.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules.d/10busybox/module-setup.sh b/modules.d/10busybox/module-setup.sh
index 179d713..9df42c8 100755
--- a/modules.d/10busybox/module-setup.sh
+++ b/modules.d/10busybox/module-setup.sh
@@ -28,6 +28,11 @@ install() {
             # See https://github.com/vda-linux/busybox_mirror/issues/33
             continue
         fi
+        if [[ ${_path##*/} == mke2fs ]]; then
+            # e2fsprogs provides mke2fs. mkfs.ext2, mkfs.ext3, and mkfs.ext4 are symlinks to it.
+            # Busybox does not provide a mkfs.ext4 applet and mkfs.ext* would point to busybox then.
+            continue
+        fi
         if [[ ${_path##*/} == mount ]]; then
             # The busybox mount applet does not resolve "auto" to the actual filesystem.
             # See https://github.com/vda-linux/busybox_mirror/issues/34
